pkg/api/read.go

```diff
@@ -68,3 +74,40 @@ func Read(reader pgmodel.Reader, metrics *Metrics) http.Handler {
 		}
 	})
 }
+
+func validateReadHeaders(w http.ResponseWriter, r *http.Request) bool {
+	// validate headers from https://github.com/prometheus/prometheus/blob/2bd077ed9724548b6a631b6ddba48928704b5c34/storage/remote/client.go
+	if r.Method != "POST" {
```

Comment:
According to our Router this also supports Get, which semantically makes sense. Lets add that as a valid option

Reply:
GET request bodies are supposed to be ignored; I think our current routing is incorrect...